### PR TITLE
Implement showing and hiding header depending on scroll

### PR DESCRIPTION
**Pull request by Frenzie**
References #7011.

Changes proposed in this pull request:

This is just a quick poc, but it shows a possible approach.

How to test the feature manually:

1. Scroll down.
2. Scroll up.
3. Repeat.

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

### CODE REVIEW

**Comment by Alkarex** [comment]
Looks good. It is already an improvement as it is. Maybe the toolbar should be included as well in what is shown when scrolling up
image

Probably related to https://github.com/FreshRSS/FreshRSS/pull/6047 but might be good while waiting for that larger PR to be completed

**Comment by Frenzie** [comment]
If you're happy with it as is I should probably fix the indentation in `app/layout/header.phtml`?

**Comment by Alkarex** [comment]
What do you think @math-GH ? It seems like a fine improvement while waiting for a more comprehensive rework

**Comment by math-GH** [comment]
Nice!

Please check:
When you scroll just 2px down, the header will disappear, but will leave an empty space:
grafik

**Comment by Frenzie** [comment]
> When you scroll just 2px down, the header will disappear, but will leave an empty space:

Adjusted by adding `currentScroll > parseInt(getComputedStyle(document.querySelector('.header')).height)`.

Edit: though if that's static maybe it can simply be stored.

File: p/scripts/main.js [hunk 1]
```diff
@@ -1954,6 +1954,23 @@ function init_confirm_action() {
 	document.querySelectorAll('button.confirm').forEach(function (b) { b.disabled = false; });
 }
 
+function init_scroll_header() {
+	const header = document.querySelector('header.header');
+	const headerHeight = parseInt(getComputedStyle(header).height);
+	if (header) {
```
Comment (Alkarex) [inline p/scripts/main.js]:
```suggestion
	if (header) {
		const headerHeight = parseInt(getComputedStyle(header).height);
```

**Comment by Alkarex** [comment]
Following up on https://github.com/FreshRSS/FreshRSS/pull/7029#issuecomment-2514400799 , I still think the toolbar should be included as well in what is shown when scrolling up. Don't you think so?

**Comment by Frenzie** [comment]
The other PR completely changes related code. As you said, it's better to wait until after.

**Comment by math-GH** [comment]
Please check the global view. The header over lays the window with the articles when a feed/category is open

**Comment by Frenzie** [comment]
scrollY becomes 0, so from the logic's perspective it's scrolled up. It's not immediately clear if there's a good way around that because you'd get the same effect from pressing home. Of course the overlay could also simply be given a higher z-index to hide the effect.

**Comment by Alkarex** [comment]
I may have merged a bit too fast here as I observe numerous UI regressions, in particular on mobile, where the toolbar appears far too often, in particular when expanding an article, but also when scrolling normally.
And on desktop using the normal view, the bar hides access to the main view:

image

So I suggest to revert and rework https://github.com/FreshRSS/FreshRSS/pull/7064

**Comment by Alkarex** [comment]
I was hopping to make a quick fix, but still not so good https://github.com/FreshRSS/FreshRSS/pull/7063

**Pull request merged**